kratos/python/add_properties_to_python.cpp

```diff
@@ -147,10 +149,37 @@ bool HasTableHelperFunction1( TContainerType& rContainer,
     return rContainer.HasTable(XVar, YVar);
 }
 
+template<typename TVariableType>
+void AddInterfaceToAccessorFold(pybind11::class_<Properties, Properties::Pointer, Properties::BaseType>& pyProperties) {
+
+    pyProperties
+    .def("GetAccessor", [](Properties &rProperties, Variable<TVariableType> &rVariable) { 
+            auto accessor = &rProperties.pGetAccessor(rVariable);
+            
+            KRATOS_ERROR_IF(*accessor == nullptr) << "Trying to get a consumed or invalid Accessor." << std::endl;
+                
+            return accessor; 
+        }, py::return_value_policy::reference_internal)
+    .def("SetAccessor", [](Properties &rProperties, Variable<TVariableType> &rVariable, std::unique_ptr<Accessor>& rpAccessor) {
+            if (rpAccessor == nullptr)
+                KRATOS_ERROR << "Trying to set a consumed or invalid Accessor. Accessors are unique. Please create a different one." << std::endl;
+
+            rProperties.SetAccessor(rVariable, std::move(rpAccessor));
+        }, py::return_value_policy::reference_internal)
+    .def("HasAccessor", [](Properties &rProperties, Variable<TVariableType> &rVariable) { 
+            return rProperties.HasAccessor(rVariable);
+        })
+    ; 
+}
+
+template<typename... TArgs>
+void AddInterfaceToAccessor(pybind11::class_<Properties, Properties::Pointer, Properties::BaseType>& pyProperties) {
```

Comment:
why do we need variadic arguments here? 

cannot we simply call this function as many times as types we have, but each time just with one template parameter?

Reply:
Because of the folding, it forwards the types you are using to instantiate to the rest of the function.